.github/workflows/ci.yml

```
@@ -333,17 +437,68 @@ jobs:
               exit 1
             fi
           done
-      - name: MCP E2E flow (initialize -> tools/list -> upload -> search)
+      - name: Verify certificate store (macOS)
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2181:style:3:6: Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?

(shellcheck)